exes/mqtt-bridge/src/spark_plug_interface.cpp

```diff
@@ -0,0 +1,342 @@
+
+      variable_metric->set_is_transient(false);
+      variable_metric->set_is_historical(false);
+    }
+
+    logger_.trace("NBIRTH payload: \n {}", payload.DebugString());
+
+    std::string payload_string;
+    payload.SerializeToString(&payload_string);
+
+    const std::string topic = topic_formatter(
+        { constants::namespace_element, config_.value().group_id, constants::nbirth, config_.value().node_id });
+
+    mqtt_client_->set_initial_message(topic, payload_string, async_mqtt::qos::at_most_once);
+
+    asio::co_spawn(mqtt_client_->strand(), mqtt_client_->send_message(topic, payload_string, async_mqtt::qos::at_most_once),
+                   asio::detached);
+  }
+}
+
+template <class config_t, class mqtt_client_t>
+auto spark_plug_interface<config_t, mqtt_client_t>::update_value(structs::signal_data& signal) -> asio::awaitable<void> {
+  const std::string signal_name = format_signal_name(signal.information.name);
+
+  Payload payload = make_payload();
+  Payload_Metric* metric = payload.add_metrics();
+
+  if (metric == nullptr) {
+    logger_.error("Failed to add metric to payload for signal: {}", signal_name);
+    co_return;
+  }
+
+  metric->set_name(signal_name);
+  metric->set_timestamp(timestamp_milliseconds().count());
+  metric->set_datatype(type_enum_convert(signal.information.type));
+
+  set_value_payload(metric, signal.current_value.value(), logger_);
+
+  logger_.trace("Updating value payload: \n {}", payload.DebugString());
+
+  std::string payload_string;
+  payload.SerializeToString(&payload_string);
+
+  logger_.trace("Sending message on topic: {}", signal_name);
+
+  co_await mqtt_client_->send_message(ndata_topic_, payload_string, async_mqtt::qos::at_most_once);
+}
+
+template <class config_t, class mqtt_client_t>
+auto spark_plug_interface<config_t, mqtt_client_t>::make_payload() -> Payload {
+  Payload payload;
+  payload.set_timestamp(timestamp_milliseconds().count());
+  if (seq_ == 255) {
+    seq_ = 0;
+  } else {
+    seq_++;
+  }
+
+  payload.set_seq(seq_);
+
+  return payload;
+}
+
+template <class config_t, class mqtt_client_t>
+auto spark_plug_interface<config_t, mqtt_client_t>::process_payload(async_mqtt::buffer const& data,
+                                                                    async_mqtt::v5::publish_packet const& publish_packet)
+    -> void {
+  Payload payload;
+
+  /// Clang-Tidy: Narrowing conversion from 'size_type' (aka 'unsigned long') to signed type 'int' is implementation-defined
+  /// data.size() is unsigned long and the parameter passed to ParseFromArray is int
+  const bool payload_valid = payload.ParseFromArray(data.data(), static_cast<int>(data.size()));
+
+  if (!payload_valid) {
+    logger_.warn("Received invalid payload. Continuing to next iteration...");
+    return;
+  }
+
+  logger_.trace("Payload parsed successfully. Processing payload...");
+
+  auto metric = payload.metrics(0);
```

Comment:
unsafe access

Reply:
Adding a check to validate that size is larger than 0